src/test/regress/expected/multi_prepare_sql.out

```diff
@@ -1195,6 +1195,31 @@ EXECUTE countsome; -- no replanning
 ---------------------------------------------------------------------
 (0 rows)
 
+-- Standard planner converted text and varchar casts to cstring in some cases
+-- We make sure we convert it back to text when parsing the expression
+-- https://github.com/citusdata/citus/issues/6061
+-- https://github.com/citusdata/citus/issues/5646
+-- https://github.com/citusdata/citus/issues/5033
+CREATE TABLE test(t timestamptz, user_id int);
+SELECT create_distributed_table('test', 'user_id');
+ create_distributed_table
+---------------------------------------------------------------------
+
+(1 row)
+
+PREPARE test_statement(text) AS
+SELECT user_id FROM test WHERE t >= $1::timestamptz ORDER BY user_id;
```

Comment:
Lets expand for different planners:
```SQL
PREPARE test_statement_regular(text) AS
SELECT user_id FROM test WHERE t >= $1::timestamptz ORDER BY user_id;


PREPARE test_statement_router(int, text) AS
SELECT user_id FROM test WHERE user_id = $1 AND t >= $2::timestamptz ORDER BY user_id;

PREPARE test_statement_repartition(int, text) AS
SELECT count(*) FROM test t1 JOIN test t2 USING (t) WHERE t1.user_id = $1 AND t >= $2::timestamptz;


PREPARE test_statement_cte(text, text) AS
with cte_1 as (SELECT user_id, t FROM test WHERE t >= $1::timestamptz ORDER BY user_id LIMIT 1) SELECT * FROM cte_1 WHERE t <=$2::timestamp ;


PREPARE test_statement_insert(int, text) AS
INSERT INTO test VALUES ($2::timestamptz, $1);
```

Reply:
Good idea, thanks for providing the prepared statements.